misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm

```diff
@@ -23,126 +24,143 @@ use strict;
 use warnings;
 use Carp;
 use DBI;
+use Text::CSV;
 
-use base qw( XrefParser::BaseParser );
-
+use parent qw( XrefParser::BaseParser );
 
 sub run {
 
-  my ($self, $ref_arg) = @_;
-  my $source_id    = $ref_arg->{source_id};
-  my $species_id   = $ref_arg->{species_id};
-  my $files        = $ref_arg->{files};
-  my $verbose      = $ref_arg->{verbose};
-  my $dbi          = $ref_arg->{dbi};
-  $dbi = $self->dbi unless defined $dbi;
-
-  if((!defined $source_id) or (!defined $species_id) or (!defined $files) ){
+  my ( $self, $ref_arg ) = @_;
+  my $source_id  = $ref_arg->{source_id};
+  my $species_id = $ref_arg->{species_id};
+  my $files      = $ref_arg->{files};
+  my $verbose    = $ref_arg->{verbose} // 0;
+  my $dbi        = $ref_arg->{dbi} // $self->dbi;
+
+  if ( ( !defined $source_id )
+    or ( !defined $species_id )
+    or ( !defined $files ) )
+  {
     croak "Need to pass source_id, species_id and file as pairs";
   }
-  $verbose |=0;
 
-  my $file = @{$files}[0];
+  my $file = shift @{$files};
 
   my %label;
   my %version;
   my %description;
   my %accession;
 
-  my $sql = 'select source_id, priority_description from source where name like "MGI"';
+  my $sql = 'select source_id from source where name like "MGI"';
```

Comment:
Could this be delegated to BaseParser::get_source_id_for_source_name() ? The like isn't helpful without wildcards anyway, it's just a slower equals.

Reply:
See also get_source_ids_for_source_name_pattern()